modules/openapi-generator/src/main/resources/go/api.mustache

```diff
@@ -356,9 +356,22 @@ func (a *{{{classname}}}Service) {{nickname}}Execute(r {{#structPrefix}}{{&class
 		{{#dataType}}
 		{{^is1xx}}
 		{{^is2xx}}
+		{{#range}}
+		{{#is3xx}}
+		if localVarHTTPResponse.StatusCode >= 300 && localVarHTTPResponse.StatusCode < 400 {
+		{{/is3xx}}
+		{{#is4xx}}
+		if localVarHTTPResponse.StatusCode >= 400 && localVarHTTPResponse.StatusCode < 500 {
+		{{/is4xx}}
+		{{#is5xx}}
+		if localVarHTTPResponse.StatusCode >= 500
+		{{/is5xx}}
+		{{/range}}
+		{{^range}}
```

Comment:
Thanks for the PR. Shall we remove the following before and after the `{{#range}} ... {{/range}}` check?
```
		{{^is1xx}}
		{{^is2xx}}

...
		{{/is2xx}}
		{{/is1xx}}
```

Reply:
Removing those checks causes unreachable code to be generated due to the `if localVarHTTPResponse.StatusCode >= 300 {` a few lines up. We're only interested in decoding "unsuccessful" responses at this point and those negative checks are the only way to do this AFAIK.

Reply:
👌